### PR TITLE
Exchange rate calculator

### PR DESCRIPTION
**Pull request by rogergros**
- Added exchange rate calculator service
- This is usefull if you only need the exchange rate (e.g. to do masive changes)

### CODE REVIEW

File: src/Elcodi/Component/Currency/Services/CurrencyConverter.php [hunk 1]
```diff
@@ -113,81 +101,14 @@ private function convertCurrency(
             return Money::create($amount, $currencyFrom);
         }
 
-        $convertedAmount = $this->convertBetweenIsos(
-            $currencyFrom->getIso(),
-            $currencyTo->getIso(),
-            $amount
+        $exchangeRate = $this->exchangeRateCalculator->calculateExchangeRate(
```
Comment (mmoreram) [inline src/Elcodi/Component/Currency/Services/CurrencyConverter.php]: Can you format this line with new lines?

Comment (rogergros) [inline src/Elcodi/Component/Currency/Services/CurrencyConverter.php]: Done!

**Comment by mmoreram** [comment]
Could you please explain us what this change is intended for? There are some important changes here!

**Comment by rogergros** [comment]
As I pointed you sometimes could need the exchange rate to do certain actions like bulk products currency change. Right now it's not possible to get the Exchange rate for a currency.

Besides this there's also a coupling with the USD (It's harcoded in some points of the code) while most of the code uses the currency parameter and not the USD (Like the currency populator). This could lead to a bit problem. E.g. I'm using the EUR as default money, I populate the DB and the populator gets all the exchange rates from EUR to all my currencies but the MoneyConvertor uses the USD (Harcoded) to exchange the money.

**Comment by mickaelandrieu** [comment]
:+1:

**Comment by mmoreram** [comment]
need some time for reviewing. Thanks @rogergros

**Comment by rogergros** [comment]
:+1:

**Pull request merged**